std/src/persistence/stock.rs

```diff
@@ -173,10 +173,15 @@ impl Stock {
         self.history.insert(id, history)?;
 
         let contract_id = consignment.contract_id();
-        self.contract_index.insert(contract_id, ContractIndex {
-            public_opouts: empty!(),
-            outpoint_opouts: empty!(),
-        })?;
+        match self.contract_index.get(&contract_id) {
+            None => {
+                self.contract_index.insert(contract_id, ContractIndex {
+                    public_opouts: empty!(),
+                    outpoint_opouts: empty!(),
+                })?;
+            }
+            Some(_) => {}
+        }
```

Comment:
In rust it will be more idiomatic to have
```suggestion
        self.contract_index.entry(contract_id).or_default();
```
it will do the same thing

Reply:
I tried to apply the suggested change but then it fails building with this error:
```
59  | pub struct ContractIndex {
    | ------------------------ doesn't satisfy `ContractIndex: std::default::Default`
...
176 |         self.contract_index.entry(contract_id).or_default();
    |                                                ^^^^^^^^^^ method cannot be called on `Entry<'_, ContractId, ContractIndex>` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `ContractIndex: std::default::Default`
help: consider annotating `ContractIndex` with `#[derive(Default)]`
    |
59  + #[derive(Default)]
60  | pub struct ContractIndex {
    |
```
and by annotating `ContractIndex` with `#[derive(Default)]` or manually implementing the `Default` trait I receive the following error:
```
error[E0119]: conflicting implementations of trait `StrictDumb` for type `ContractIndex`
  --> /rgb-wallet/std/src/persistence/stock.rs:57:22
   |
57 | #[derive(StrictType, StrictDumb, StrictEncode, StrictDecode)]
   |                      ^^^^^^^^^^
   |
   = note: conflicting implementation in crate `strict_encoding`:
           - impl<T> StrictDumb for T
             where T: StrictType, T: std::default::Default;
   = note: this error originates in the derive macro `StrictDumb` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Yes, `Entry` API doens't (yet) work with confined collections from `amplify`, that was my bad.

I did it in a slightly different way - see my changes in the pushed commits

Reply:
I've checked, changes work, thanks